### PR TITLE
[FLINK-8329] [flip6] Move YarnClient to AbstractYarnClusterDescriptor

### PR DESCRIPTION
**Pull request by tillrohrmann**
## What is the purpose of the change

Moves the YarnClient from the YarnClusterClient to the AbstractYarnClusterDescriptor.
This makes the latter responsible for the lifecycle management of the client and gives
a better separation of concerns.

This PR is based on #5215 

## Brief change log

- Move the `YarnClient` from the `YarnClusterClient` to the `AbstractYarnClusterDescriptor`

## Verifying this change

This change is already covered by existing tests

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by GJL** [comment]
There are some checkstyle violations:
```
[ERROR] src/main/java/org/apache/flink/yarn/cli/YarnApplicationStatusMonitor.java:[33,8] (imports) UnusedImports: Unused import: java.util.concurrent.ScheduledExecutorService.
[ERROR] src/main/java/org/apache/flink/yarn/YarnClusterClient.java:[46,8] (imports) UnusedImports: Unused import: org.apache.hadoop.yarn.client.api.YarnClient.
```

**Comment by tillrohrmann** [comment]
Corrected the checkstyle violations.

File: flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNITCase.java [hunk 1]
```diff
@@ -78,6 +83,9 @@ public void testPerJobMode() {
 		jobGraph.addJar(new org.apache.flink.core.fs.Path(testingJar.toURI()));
 
 		YarnClusterClient clusterClient = yarnClusterDescriptorV2.deployJobCluster(clusterSpecification, jobGraph);
+
+		clusterClient.shutdown();
+		yarnClusterDescriptorV2.close();
```
Comment (GJL) [inline flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNITCase.java]: Closing in finally or try-with-resource not needed?

File: flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNSessionFIFOITCase.java [hunk 2]
```diff
@@ -272,6 +275,7 @@ public void testJavaAPI() throws Exception {
 		LOG.info("Shutting down cluster. All tests passed");
 		// shutdown cluster
 		yarnCluster.shutdown();
+		clusterDescriptor.close();
```
Comment (GJL) [inline flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNSessionFIFOITCase.java]: Closing in finally or try-with-resource not needed?

File: flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java [hunk 3]
```diff
@@ -143,31 +138,48 @@
...
+			String configurationDirectory,
+			YarnClient yarnClient) {
+
+		yarnConfiguration = new YarnConfiguration();
+
 		// for unit tests only
 		if (System.getenv("IN_TESTS") != null) {
 			try {
-				conf.addResource(new File(System.getenv("YARN_CONF_DIR") + "/yarn-site.xml").toURI().toURL());
+				yarnConfiguration.addResource(new File(System.getenv("YARN_CONF_DIR") + "/yarn-site.xml").toURI().toURL());
```
Comment (GJL) [inline flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java]: I think it's safer to use `new File(System.getenv("YARN_CONF_DIR"),  "yarn-site.xml")`

File: flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNITCase.java (hunk 1 above)
Comment (tillrohrmann) [reply flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNITCase.java]: True will refactor it.

File: flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNSessionFIFOITCase.java (hunk 2 above)
Comment (tillrohrmann) [reply flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNSessionFIFOITCase.java]: will refactor it

File: flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java (hunk 3 above)
Comment (tillrohrmann) [reply flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java]: True, will change it.

**Pull request closed without merging**